ztools/xssh/kex.go

```diff
@@ -165,7 +165,7 @@ type dhGroup struct {
 	JsonLog       dhGroupJsonLog
 }
 type dhGroupJsonLog struct {
-	Parameters      *ztoolsKeys.DHParams  `json:"params,omitempty"`
+	Parameters      *ztoolsKeys.DHParams  `json:"dh_params,omitempty"`
```

Comment:
I'm a little confused by how this fits together. I was expecting something along the lines of:
```go
type KeyExchange struct {
    DHParams *DHParams `json:"dh_params,omitempty"`
    ECDHParams *ECDHParams `json:"ecdh_params,omitempty"`
    Ed25519Params *Ed25519Params `json:"ed25519_params,omitempty"`
}

type ServerHostKey struct {
    RSAPublic `json:"rsa_public_key,omitempty"`
    ECDHPublic `json:"ecdh_public_key,omitempty"`
    Ed25519Public `json:"ed25519_public_key,omitmepty"`
}

type SSH struct {
    KeyExchange *KeyExchange `json:"key_exchange,omitempty"`
    ServerHostKey *ServerHostKey `json:"server_host_key,omitempty"`
    ServerSignature *ServerSignature `json:"server_signature,omitempty"`
}
```

Why are key exchange types duplicating fields?